clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -381,10 +417,10 @@ public void testEmptyBatchDeletion() {
                 ByteBuffer filtered = ByteBuffer.allocate(2048);
                 MemoryRecords records = MemoryRecords.readableRecords(buffer);
                 MemoryRecords.FilterResult filterResult = records.filterTo(new TopicPartition("foo", 0),
-                        new MemoryRecords.RecordFilter() {
+                        new MemoryRecords.RecordFilter(0, 0) {
                             @Override
-                            protected BatchRetention checkBatchRetention(RecordBatch batch) {
-                                return deleteRetention;
+                            protected BatchRetentionResult checkBatchRetention(RecordBatch batch) {
+                                return new BatchRetentionResult(deleteRetention, true);
```

Comment:
Since there is no marker, It seems that containsMarkerForEmptyTxn should be false.